ironic.conf.j2

```diff
@@ -66,7 +66,13 @@ send_sensor_data_interval = 160
 bootloader = http://{{ env.IRONIC_URL_HOST }}:{{ env.HTTP_PORT }}/uefi_esp.img
 
 [database]
-connection = mysql+pymysql://ironic:{{ env.MARIADB_PASSWORD }}@localhost/ironic?charset=utf8
+{% if env.MARIADB_TLS_ENABLED == "true" %}
+connection = mysql+pymysql://ironic:{{ env.MARIADB_PASSWORD }}@127.0.0.1/ironic?charset=utf8&ssl=on&ssl_ca={{ env.MARIADB_CACERT_FILE }}
```

Comment:
Do we need to explicitly set `ssl_verify_cert`?

Reply:
I think the option `ssl_verify_server_cert` is required if we use mTLS, which is not in our case.

Reply:
I see. That is fixed! Thanks

Reply:
I have tested the use of `ssl_verify_server_cert`. Interestingly, we cannot add this option because Ironic denies this parameter. It seems like Ironic doesn't directly pass the parameter to the mariaDB client configuration. However, just `ssl=on` is enough since SQLAlchemy by default verifies the certificate. This behaviour has been tested. I changed the target IP address to `127.0.1.1` and this is the error:
```
ssl.CertificateError: hostname '127.0.1.1' doesn't match '127.0.0.1'
: ssl.CertificateError: hostname '127.0.1.1' doesn't match '127.0.0.1'
```

Reply:
Ah, I had been looking at the docs for the [python connector for mysql](https://dev.mysql.com/doc/connector-python/en/connector-python-connectargs.html), where the option is called `ssl_verify_cert` and defaults to False.

However, what I failed to realise is that we are actually using pymysql, and [in pymysql](https://github.com/PyMySQL/PyMySQL/blob/v0.10.1/pymysql/connections.py#L336) the option is called `check_hostname` and it defaults to True if a CA is specified (which is the only sane way of implementing SSL). Good to see that testing confirms this works.

FWIW the `ssl=on` part [doesn't appear to do anything](https://github.com/sqlalchemy/sqlalchemy/blob/rel_1_3_20/lib/sqlalchemy/dialects/mysql/mysqldb.py#L203-L211).

Reply:
Even if ssl=on does not do anything, I think it can be valuable to have it here since it makes it clear that we are configuring the connection with ssl for whoever reads this configuration file